pkg/ovs/openflow/ofctrl_bridge_test.go

```diff
@@ -160,3 +162,43 @@ func TestOFBridgePacketRcvd(t *testing.T) {
 		}
 	}
 }
+
+func TestOFMeterStats(t *testing.T) {
+	// Reset OVS metrics (Prometheus) and reinitialize them to test.
+	legacyregistry.Reset()
+	metrics.InitializeOVSMetrics()
+
+	b := NewOFBridge("test-br", GetMgmtAddress(ovsconfig.DefaultOVSRunDir, "test-br"))
+	sw := newFakeOFSwitch(b)
+	b.SetOFSwitch(sw)
+
+	mpMeterStatsReply := openflow15.NewMpReply(openflow15.MultipartType_MeterStats)
+	newMeterStats := func(meterID uint32, packets uint64) *openflow15.MeterStats {
+		meterStats := openflow15.NewMeterStats(meterID)
+		bandStats := openflow15.NewMeterBandStats()
+		bandStats.PacketBandCount = packets
+		meterStats.AddBandStats(*bandStats)
+		return meterStats
+	}
+	meterStats1 := newMeterStats(1, 100)
+	meterStats2 := newMeterStats(2, 0)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats1)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats2)
+	checkMeterStatsReply := func(meterId int, packetCounts int64) {
+		switch meterId {
+		case 1:
+			assert.Equal(t, packetCounts, int64(100))
+		case 2:
+			assert.Equal(t, packetCounts, int64(0))
+		}
+	}
+	b.GetMeterStats(checkMeterStatsReply)
+	// To make reply has the same Xid as request which is constructed with GetMeterStats
+	b.mpReplyChsMutex.RLock()
+	for k := range b.mpReplyChs {
+		mpMeterStatsReply.Xid = k
+	}
+	b.mpReplyChsMutex.RUnlock()
+	b.MultipartReply(sw, mpMeterStatsReply)
+	time.Sleep(time.Second)
```

Comment:
This is nondeterministic and may be flaky. And having the assertion in the callback can't guarantee it will be called. Even if the tested function does nothing, the test would't fail.

Perhaps you can use a map to collect the received stats and use `assert.Eventually` to check whether expected stats are received in a reasonable time.